components/uikit/Input/BaseInput.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import {
+  Box,
+  Input,
+  InputField,
+  InputSlot,
+  FormControl,
+  FormControlError,
+  FormControlErrorText,
+  FormControlHelper,
+  FormControlHelperText,
+} from "@gluestack-ui/themed";
+
+export type BaseInputProps = {
+  placeholder: string;
+  error?: string;
+  help?: string;
+  isRequired?: boolean;
+  icon?: React.ElementType;
+  value?: string;
+  onChange?: (value: string) => void;
+  type?: "text" | "password" | undefined;
```

Comment:
le `undefined` n'est pas nécessaire, car le `?` juste apres `type` fait signifier qu'il peut être `undefined`
```suggestion
  type?: "text" | "password";
```